Mathlib/Probability/ConditionalProbability.lean

```diff
@@ -161,48 +161,84 @@ theorem cond_pos_of_inter_ne_zero [IsFiniteMeasure μ]
+  obtain hcs | hcs := eq_or_ne (μ s) 0
+  · simp [hcs, measure_inter_null_of_null_left]
+  · rw [cond_apply μ hms t, mul_comm, ← mul_assoc, ENNReal.mul_inv_cancel hcs hcs', one_mul]
 #align probability_theory.cond_mul_eq_inter' ProbabilityTheory.cond_mul_eq_inter'
 
-theorem cond_mul_eq_inter [IsFiniteMeasure μ] (hms : MeasurableSet s) (hcs : μ s ≠ 0) (t : Set Ω) :
-    μ[t|s] * μ s = μ (s ∩ t) :=
-  cond_mul_eq_inter' μ hms hcs (measure_ne_top _ s) t
+theorem cond_mul_eq_inter [IsFiniteMeasure μ] (hms : MeasurableSet s) (t : Set Ω) :
+    μ[t|s] * μ s = μ (s ∩ t) := cond_mul_eq_inter' μ hms (measure_ne_top _ s) t
 #align probability_theory.cond_mul_eq_inter ProbabilityTheory.cond_mul_eq_inter
 
 /-- A version of the law of total probability. -/
-theorem cond_add_cond_compl_eq [IsFiniteMeasure μ] (hms : MeasurableSet s) (hcs : μ s ≠ 0)
-    (hcs' : μ sᶜ ≠ 0) : μ[t|s] * μ s + μ[t|sᶜ] * μ sᶜ = μ t := by
-  rw [cond_mul_eq_inter μ hms hcs, cond_mul_eq_inter μ hms.compl hcs', Set.inter_comm _ t,
+theorem cond_add_cond_compl_eq [IsFiniteMeasure μ] (hms : MeasurableSet s) :
+    μ[t|s] * μ s + μ[t|sᶜ] * μ sᶜ = μ t := by
+  rw [cond_mul_eq_inter μ hms, cond_mul_eq_inter μ hms.compl, Set.inter_comm _ t,
     Set.inter_comm _ t]
   exact measure_inter_add_diff t hms
 #align probability_theory.cond_add_cond_compl_eq ProbabilityTheory.cond_add_cond_compl_eq
 
 /-- **Bayes' Theorem** -/
 theorem cond_eq_inv_mul_cond_mul [IsFiniteMeasure μ]
     (hms : MeasurableSet s) (hmt : MeasurableSet t) : μ[t|s] = (μ s)⁻¹ * μ[s|t] * μ t := by
-  by_cases ht : μ t = 0
-  · simp [cond, ht, Measure.restrict_apply hmt, Or.inr (measure_inter_null_of_null_left s ht)]
-  · rw [mul_assoc, cond_mul_eq_inter μ hmt ht s, Set.inter_comm, cond_apply _ hms]
+  rw [mul_assoc, cond_mul_eq_inter μ hmt s, Set.inter_comm, cond_apply _ hms]
 #align probability_theory.cond_eq_inv_mul_cond_mul ProbabilityTheory.cond_eq_inv_mul_cond_mul
 
 end Bayes
 
+lemma comap_cond {i : Ω' → Ω} (hi : MeasurableEmbedding i) (hi' : ∀ᵐ ω ∂μ, ω ∈ range i)
+    (hs : MeasurableSet s) : comap i (μ[|s]) = (comap i μ)[|i ⁻¹' s] := by
+  ext t ht
+  change μ (range i)ᶜ = 0 at hi'
+  rw [cond_apply, comap_apply, cond_apply, comap_apply, comap_apply, image_inter,
+    image_preimage_eq_inter_range, inter_right_comm, measure_inter_conull hi',
+    measure_inter_conull hi']
+  all_goals first
+  | exact hi.injective
+  | exact hi.measurableSet_image'
+  | exact hs
+  | exact ht
+  | exact hi.measurable hs
+  | exact (hi.measurable hs).inter ht
+
+variable [Fintype α] [MeasurableSpace α] [DiscreteMeasurableSpace α]
+
+/-- The **law of total probability**: a measure `μ` can be expressed as a mixture of its conditional
+measures `μ[|Y ← y]` from a random variable `Y` valued in a fintype. -/
+lemma law_of_total_probability {Y : Ω → α} (hY : Measurable Y) (μ : Measure Ω) [IsFiniteMeasure μ] :
+    μ = ∑ y, μ (Y ⁻¹' {y}) • (μ[|Y ← y]) := by
```

Comment:
Can you revert this equality, and give a more descriptive name to the theorem, mentioning sum and cond?

I also don't like that name because this is not the only statement that could be called "law of total probability". For example, when I think of that law, I imagine the first of the two equalities on the wikipedia page: https://en.wikipedia.org/wiki/Law_of_total_probability . There is also the continuous case...

Reply:
What about now?